.github/workflows/publish-alloy-devel.yml

```diff
@@ -112,6 +101,12 @@ jobs:
         echo "$(bash ./tools/image-tag-docker)" > .tag-only
         echo "grafana/alloy-dev:$(bash ./tools/image-tag-docker)" > .image-tag
 
+    - name: Log in to Google Artifact Registry
```

Comment:
```suggestion
      # This step needs to run after "Checkout code".
      # That's because the login to GAR generates a new file.
      # We don't want this file to end up in the repo directory.
      # Then "tools/image-tag" would get confused because "git status" no longer reports a clean repo.
    - name: Log in to Google Artifact Registry
```